internal/rest/dequeue_nodes.go

```diff
@@ -724,7 +724,9 @@ func (rest *RestOperations) RefreshCacheForRetryLayer(parentVsKey string, aviObj
 				case avimodels.Pool:
 					poolObjName = *rest_op.Obj.(avimodels.Pool).Name
 				}
-				rest_op.ObjName = poolObjName
+				if poolObjName != "" {
```

Comment:
When would poolObjName be empty ?

Reply:
@monotosh-avi : When we do delete call , we do not fill up rest_op.obj object so rest_op.Obj is nil... in that case switch will not be executed. This will result in empty poolObjName. So when AKO will try to remove this entry from cache of Pool.. but as pool object name is empty, it will not be removed from cache. This will trigger loop of DELETE calls till retry count.  Same logic applicable for other objects. I have seen repeated DELETE call when i did out of band secure VS deletion and afterward ingress updation for path.

Reply:
Good catch !